pinot-query-planner/src/main/java/org/apache/pinot/query/catalog/PinotCatalog.java

```diff
@@ -80,8 +82,20 @@ public Table getTable(String name) {
    */
   @Override
   public Set<String> getTableNames() {
-    return _tableCache.getTableNameMap().keySet().stream().filter(n -> DatabaseUtils.isPartOfDatabase(n, _databaseName))
-        .collect(Collectors.toSet());
+    if (StringUtils.isBlank(_databaseName)) {
+      return _tableCache.getTableNameMap().keySet().stream()
+          .filter(n -> DatabaseUtils.isPartOfDatabase(n, _databaseName))
+          .collect(Collectors.toSet());
+    } else {
+      Set<String> result = new HashSet<>();
+      for (String n: _tableCache.getTableNameMap().keySet()) {
```

Comment:
nit: let's use a more descriptive name like `tableName` or `fullyQualifiedTableName` instead of `n`?

Reply:
Done (and also unified both default and non-default scenarios after last modifications):

https://github.com/apache/pinot/pull/14830/commits/115908e1953cd4690e909307a441006c5ac0bd7f